extras-scalapb/build.gradle

```diff
@@ -0,0 +1,40 @@
+apply plugin: 'com.google.protobuf'
+
+archivesBaseName = "rabbitmq-client-extras-scalapb_$scalaVersion"
+
+dependencies {
+    compile project(":core")
+
+    compile "com.thesamet.scalapb:scalapb-runtime_$scalaVersion:$scalapbVersion"
+    compile "com.thesamet.scalapb:scalapb-json4s_$scalaVersion:$scalapbJson4sVersion"
```

Comment:
why not https://github.com/scalapb-json/scalapb-circe , since we always work with Circe (and not json4s)?

Reply:
I had two reasons:
1. json4s is supported directly by the ScalaPB, it is not an _external_ module. https://scalapb.github.io/docs/json
1. (not so important) We use circe everywhere so another usage could cause more classpath clashes, so json4s seems to be safer alternative for me.

Reply:
`scalapb-circe` seems well maintained/updated and IMHO it's better to use the libraries that we actually use (if you see what I mean :D ), because it helps with interoperability (with things that may not even be related to RabbitMQ and this `rabbitmq-scala-client` library)
let's see what @jendakol has to say about it

Reply:
This PR doesn't introduce any json4s specific type to the public API, the JSON library used under the hood is completelly hidden from the caller.

As scalapb-json4s is maintained by the same person as scalapb itselt, it is better choice for me (than relying on a 3rd party library).

Reply:
I don't know about the binary converters (`ScalaPBAsBinaryProductConverter` & co.), but given that there is already `JsonDeliveryConverter` and `JsonProductConverter` in `extras-circe`, we may not need the ad-hoc JSON convertors (`ScalaPBAsJsonProductConverter` & co.) after all.

For example:

`JsonDeliveryConverter` on its own can create Rabbit `ProductConverter` for anything, that has a Circe `Encoder` available. That's where `scalapb-circe` comes into play, it can "automagically" create `Encoder` for any `GeneratedMessage` (from ScalaPB), or you can customize the JSON Printer which is probably what we would need.
Then the remaining question is, where such canonical Encoder for `GeneratedMessage`s should live? We could put it here (and introduce a dependency on `scalapb-circe`), but I don't think that makes much sense, because there's nothing related to RabbitMQ in it. So we could have such 
```scala
implicit def encoder[A <: scalapb.GeneratedMessage]: io.circe.Encoder[A]
```
in our internal Service Utils. I think that would be a better design, because it would be more orthogonal and reusing already existing parts. What do you think?